src/dex/jarvis-v6/jarvis-v6.ts

```diff
@@ -0,0 +1,463 @@
+      poolState,
+    );
+
+    return [
+      {
+        prices,
+        unit,
+        data: {
+          swapFunction,
+          poolAddress,
+        },
+        poolAddresses: [poolAddress],
+        exchange: this.dexKey,
+        gasCost: 475 * 1000, //between 450-500k gas
+        poolIdentifier,
+      },
+    ];
+  }
+
+  // Encode params required by the exchange adapter
+  // Used for multiSwap, buy & megaSwap
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { swapFunction } = data;
+    const type = [JarvisSwapFunctions.MINT, JarvisSwapFunctions.REDEEM].indexOf(
+      swapFunction,
+    );
+
+    if (type === undefined) {
+      throw new Error(
+        `Jarvis: Invalid OpType ${swapFunction}, Should be one of ['mint', 'redeem']`,
+      );
+    }
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          opType: 'uint',
+          destPool: 'address',
+          expiration: 'uint128',
+        },
+      },
+      {
+        opType: type,
+        destPool: data.poolAddress.toLowerCase() || NULL_ADDRESS,
+        expiration: (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0),
+      },
+    );
+
+    return {
+      targetExchange: data.poolAddress.toLowerCase(),
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // Encode call data used by simpleSwap like routers
+  // Used for simpleSwap & simpleBuy
+  // Hint: this.buildSimpleParamWithoutWETHConversion
+  // could be useful
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: JarvisV6Data,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const { swapFunction } = data;
+    const timestamp = (Date.now() / 1000 + THIRTY_MINUTES).toFixed(0);
+
+    let swapFunctionParams: JarvisV6Params;
+    switch (swapFunction) {
+      case JarvisSwapFunctions.MINT:
+        swapFunctionParams = [
+          destAmount,
+          srcAmount,
+          timestamp,
+          this.augustusAddress,
+        ];
+        break;
+      case JarvisSwapFunctions.REDEEM:
+        swapFunctionParams = [
+          srcAmount,
+          destAmount,
+          timestamp,
+          this.augustusAddress,
+        ];
+        break;
+      default:
+        throw new Error(`Unknown function ${swapFunction}`);
+    }
+    const swapData = this.poolInterface.encodeFunctionData(swapFunction, [
+      swapFunctionParams,
+    ]);
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      data.poolAddress.toLowerCase(),
+    );
+  }
+
+  // Returns list of top pools based on liquidity. Max
+  // limit number pools should be returned.
+  async getTopPoolsForToken(
```

Comment:
This should be implemented to help our multipath engine use jFiats as intermediary tokens. 
Also, as the protocol doesn't have `exchange()` anymore, I believe this will allow atomic swaps between jFiats.

Reply:
Cf. comment above:  another contract will be deployed to handle the swaps between jFiats